exchanges/poloniex/poloniex_websocket_trade.go

```diff
@@ -0,0 +1,109 @@
+		return nil, err
+	}
+	var resp []*OrderIDResponse
+	if err := e.SendWebsocketRequest(ctx, "createOrder", arg, &resp); err != nil {
+		return nil, err
+	}
+	if len(resp) != 1 {
+		return nil, common.ErrInvalidResponse
+	} else if resp[0].Code != 0 && resp[0].Code != 200 {
+		return nil, fmt.Errorf("%w: error code: %d message: %s", order.ErrPlaceFailed, resp[0].Code, resp[0].Message)
+	}
+	return resp[0], nil
+}
+
+// WsCancelMultipleOrdersByIDs batch cancel one or many active orders in an account by IDs through the websocket stream.
+func (e *Exchange) WsCancelMultipleOrdersByIDs(ctx context.Context, orderIDs, clientOrderIDs []string) ([]*WsCancelOrderResponse, error) {
+	if len(clientOrderIDs) == 0 && len(orderIDs) == 0 {
+		return nil, order.ErrOrderIDNotSet
+	}
+	params := make(map[string][]string)
+	if len(clientOrderIDs) > 0 {
+		params["clientOrderIds"] = clientOrderIDs
+	}
+	if len(orderIDs) > 0 {
+		params["orderIds"] = orderIDs
+	}
+	var resp []*WsCancelOrderResponse
+	err := e.SendWebsocketRequest(ctx, "cancelOrders", params, &resp)
+	if err != nil {
+		return nil, err
+	}
+	for r := range resp {
+		if resp[r].Code != 0 && resp[r].Code != 200 {
+			err = common.AppendError(err, fmt.Errorf("%w: code: %d message: %s", order.ErrCancelFailed, resp[r].Code, resp[r].Message))
+		}
+	}
+	return resp, err
+}
+
+// WsCancelTradeOrders batch cancel all orders in an account.
+func (e *Exchange) WsCancelTradeOrders(ctx context.Context, symbols []string, accountTypes []AccountType) ([]*WsCancelOrderResponse, error) {
+	args := make(map[string]any)
+	if len(symbols) > 0 {
+		args["symbols"] = symbols
+	}
+	if len(accountTypes) > 0 {
+		args["accountTypes"] = accountTypes
+	}
+	var resp []*WsCancelOrderResponse
+	err := e.SendWebsocketRequest(ctx, "cancelAllOrders", args, &resp)
+	if err != nil {
+		return nil, err
+	}
+	for r := range resp {
+		if resp[r].Code != 0 && resp[r].Code != 200 {
+			err = common.AppendError(err, fmt.Errorf("%w: code: %d message: %s", order.ErrCancelFailed, resp[r].Code, resp[r].Message))
+		}
+	}
+	return resp, err
+}
+
+// SendWebsocketRequest sends a websocket request through the private connection
+func (e *Exchange) SendWebsocketRequest(ctx context.Context, event string, arg, response any) error {
+	conn, err := e.Websocket.GetConnection(connSpotPrivate)
+	if err != nil {
+		return err
+	}
+	input := &struct {
+		ID     string `json:"id"`
+		Event  string `json:"event"`
+		Params any    `json:"params"`
+	}{
+		ID:     e.MessageID(),
+		Event:  event,
+		Params: arg,
+	}
+	e.spotSubMtx.Lock()
+	result, err := conn.SendMessageReturnResponse(ctx, sWebsocketPrivateEPL, input.ID, input)
+	e.spotSubMtx.Unlock()
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal(result, &WebsocketResponse{Data: response}); err != nil {
+		return err
+	}
+	if response == nil {
+		return common.ErrNoResponse
+	}
+	if errType, ok := response.(interface{ Error() error }); ok && errType.Error() != nil {
```

Comment:
🚧 Change request:
Erm. What? 👀 
This doesn't make sense, none of the response types have an Error method.

But 🎉  They could.

I suggest you cast to `[]interface{ Error() error }` instead, and then range that mofo.

But also:
🚧 Change request:
We're using this anonymous type a few times.
Can we have a local `type hasError` ?
Or maybe if we do it elsewhere in `common.HasError` since okx does the same thing.